Gems/ScriptCanvasDeveloper/Code/Editor/Source/Tools/TranslationBrowser/TranslationBrowser.cpp

```diff
@@ -0,0 +1,897 @@
+
+        //VerificationSet verificationSet;
+        GraphCanvas::TranslationKey translationKey;
+        AZStd::vector<AZ::TypeId> nodes;
+
+        auto getNodeClasses = [&serializeContext, &nodes](const AZ::SerializeContext::ClassData*, const AZ::Uuid& type)
+        {
+
+            bool foundBaseClass = false;
+            auto baseClassVisitorFn = [&nodes, &type, &foundBaseClass](const AZ::SerializeContext::ClassData* reflectedBase, const AZ::TypeId& /*rttiBase*/)
+            {
+                if (!reflectedBase)
+                {
+                    foundBaseClass = false;
+                    return false; // stop iterating
+                }
+
+                foundBaseClass = (reflectedBase->m_typeId == azrtti_typeid<ScriptCanvas::Node>());
+                if (foundBaseClass)
+                {
+                    nodes.push_back(type);
+                    return false; // we have a base, stop iterating
+                }
+
+                return true; // keep iterating
+            };
+
+            AZ::EntityUtils::EnumerateBaseRecursive(serializeContext, baseClassVisitorFn, type);
+
+            return true;
+        };
+
+        serializeContext->EnumerateAll(getNodeClasses);
+
+        for (auto& node : nodes)
+        {
+            if (const AZ::SerializeContext::ClassData* classData = serializeContext->FindClassData(node))
+            {
+                AZStd::string cleanName = GraphCanvas::TranslationKey::Sanitize(classData->m_name);
+
+                // Tooltip attribute takes priority over the edit data description
+                AZStd::string tooltip = GraphCanvasAttributeHelper::GetStringAttribute(classData, AZ::Script::Attributes::ToolTip);
+
+                // Find the category
+                //details.m_category = GraphCanvasAttributeHelper::GetStringAttribute(classData, AZ::Script::Attributes::Category);
+                if (/*details.m_category.empty() && */classData->m_editData)
+                {
+                    auto elementData = classData->m_editData->FindElementData(AZ::Edit::ClassElements::EditorData);
+                    const AZStd::string categoryAttribute = GraphCanvasAttributeHelper::ReadStringAttribute(elementData->m_attributes, AZ::Script::Attributes::Category);
+                    if (!categoryAttribute.empty())
+                    {
+                        //details.m_category = categoryAttribute;
+                    }
+                }
+
+                if (ScriptCanvas::Node* nodeComponent = reinterpret_cast<ScriptCanvas::Node*>(classData->m_factory->Create(classData->m_name)))
+                {
+                    nodeComponent->Configure();
+
+                    [[maybe_unused]] int inputIndex = 0;
+                    [[maybe_unused]] int outputIndex = 0;
+
+                    const auto& allSlots = nodeComponent->GetAllSlots();
+                    for ([[maybe_unused]] const auto& slot : allSlots)
```

Comment:
cut loop?

Reply:
I will very likely cut this entire tool, it's purpose was to migrate the old format to the new one.